### PR TITLE
Added the parent and teacher mode for the user teacher

### PR DESCRIPTION
**Pull request by Nikhil77sutara**

### CODE REVIEW

File: src/components/selectMode/SelectModeButton.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    width: 35%;
+    height: 10vh;
+    background-color: white;
+    border-radius: min(20px, 5vh);
+    display: flex;
+    padding: 0% min(20px, 2%);
+    justify-content: space-between;
+    align-items: center;
+    cursor: pointer;
+    min-width: 170px;
```
Comment (Sutara-Skanda) [inline src/components/selectMode/SelectModeButton.css:12]: Don't use px

File: src/pages/SelectMode.tsx [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+  };
+
+  const onTeacherSelect = () => {
+    api.currentMode = MODES.TEACHER;
+    // history.replace(PAGES.SELECT_SCHOOL);
+    setStage(STAGES.SCHOOL);
+  };
+  const displayClasses = async () => {
+    const element = await api.getClassesForSchool(currentSchool, currentUser);
+    console.log("THis are the classes " + element);
```
Comment (Sutara-Skanda) [inline src/pages/SelectMode.tsx:100]: Change the **THis** word to **This**

Comment (Nikhil77sutara) [reply src/pages/SelectMode.tsx:100]: sure

**Review by Sutara-Skanda — CHANGES_REQUESTED** [review]
Fix the Requested Changes

**Review by Sutara-Skanda — APPROVED** [review]

File: src/pages/DisplayStudents.tsx [hunk 3]
```diff
@@ -11,6 +11,8 @@ import { ServiceConfig } from "../services/ServiceConfig";
 import { t } from "i18next";
 import { Util } from "../utility/util";
 import ParentalLock from "../components/parent/ParentalLock";
+import { FirebaseApi } from "../services/api/FirebaseApi";
+import { FirebaseAuth } from "../services/auth/FirebaseAuth";
```
Comment (bhanugummana) [inline src/pages/DisplayStudents.tsx:15]: Don't use FirebaseApi and FirebaseAuth directly

File: src/pages/DisplayStudents.tsx [hunk 4]
```diff
@@ -31,6 +33,21 @@ const DisplayStudents: FC<{}> = () => {
...
 
     // setStudents([...students, students[0]]);
+
+    const currentUser = await FirebaseAuth.getInstance().getCurrentUser();
+    console.log(
+      "🚀 ~ file: DisplayStudents.tsx:35 ~ getStudents ~ FirebaseAuth.getInstance().currentUser:",
+      currentUser
+    );
+    const iseTeacher = await FirebaseApi.getInstance().isUserTeacher(
+      currentUser!
```
Comment (bhanugummana) [inline src/pages/DisplayStudents.tsx:43]: don't use !

File: src/pages/SelectMode.css [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+  
+  .class-avatar-counter{
+    width: 10vw;
+    height: 10vw;
+    border-radius: 100%;
+    background-color: white;
+    color: black;
+    /* font-weight: bold; */
+    font-size: 10vh;
```
Comment (bhanugummana) [inline src/pages/SelectMode.css:111]: use clamp for font-size

File: src/pages/SelectMode.tsx [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+
+  var count = 1;
+  let tempSchoolList: {
+    id: string;
+    displayName: string;
+  }[] = [];
+  useEffect(() => {
+    init();
+  }, []);
+  const api = FirebaseApi.getInstance();
```
Comment (bhanugummana) [inline src/pages/SelectMode.tsx:52]: don't use FirebaseApi directly

**Review by bhanugummana — CHANGES_REQUESTED** [review]

**Review by bhanugummana — APPROVED** [review]

**Pull request merged**